src/tools/manifest2proto.ts

```diff
@@ -21,35 +21,47 @@ import {Manifest} from '../runtime/manifest.js';
 import {Capabilities} from '../runtime/capabilities.js';
 
 // Import Proto declaration.
-const rootNamespace = protobuf.loadSync('./java/arcs/core/data/proto/manifest.proto');
+const rootNamespace = protobuf.loadSync('java/arcs/core/data/proto/manifest.proto');
 const manifestProto = rootNamespace.lookupType('arcs.ManifestProto');
 const FATE_ENUM = rootNamespace.lookupEnum('arcs.Fate');
 const CAPABILITY_ENUM = rootNamespace.lookupEnum('arcs.Capability');
 const DIRECTION_ENUM = rootNamespace.lookupEnum('arcs.Direction');
 const PRIMITIVE_TYPE_ENUM = rootNamespace.lookupEnum('arcs.PrimitiveTypeProto');
 
-export async function serialize2proto(path: string): Promise<Uint8Array> {
+export async function encodeManifestToProto(path: string): Promise<Uint8Array> {
   const manifest = await Runtime.parseFile(path);
 
   if (manifest.imports.length) {
     throw Error('Only single-file manifests are currently supported');
   }
+  return encodePayload(await manifestToProtoPayload(manifest));
+}
 
-  const json = await manifestToProtoPayload(manifest);
-
-  const error = manifestProto.verify(json);
-  if (error) throw Error(error);
+export async function manifestToProtoPayload(manifest: Manifest) {
+  return makeManifestProtoPayload(manifest.particles, manifest.recipes);
+}
 
-  return manifestProto.encode(manifestProto.create(json)).finish();
+export async function encodePlansToProto(plans: Recipe[]) {
+  const specMap = new Map<string, ParticleSpec>();
+  for (const spec of [].concat(...plans.map(r => r.particles)).map(p => p.spec)) {
```

Comment:
Took me a little bit to understand what `[].concat(...)` was doing. I guess it's OK, but I'd have just used nested for-loops myself.

Reply:
flatMap will arrive somday... `[].concat(...)` seems quite canonical in JS community to do array flattening.

Reply:
MDN docs use a reduce with concat to implement flatmap, see Alternative here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap